src/python/pants/backend/python/subsystems/python_tool_base.py

```diff
@@ -10,11 +10,12 @@ class PythonToolBase(Subsystem):
     """Base class for subsystems that configure a python tool to be invoked out-of-process."""
 
     # Subclasses must set.
-    default_version: Optional[str] = None
-    default_entry_point: Optional[str] = None
+    default_version: ClassVar[Optional[str]] = None
+    default_entry_point: ClassVar[Optional[str]] = None
     # Subclasses do not need to override.
-    default_extra_requirements: Sequence[str] = []
-    default_interpreter_constraints: Sequence[str] = []
+    default_extra_requirements: ClassVar[Sequence[str]] = []
+    default_interpreter_constraints: ClassVar[Sequence[str]] = []
+    register_interpreter_constraints: ClassVar[bool] = True
```

Comment:
Should this default to False? It seems like defaulting to limiting the total number of interpreters that are in use by only giving tools that _require_ the ability to use an interpreter other than the one the code is using would be good.